build.gradle

```diff
@@ -701,19 +804,36 @@ def deobf(String sourceURL) {
 }
 
 // The method above is to be preferred. Use this method if the filename is not at the end of the URL.
-def deobf(String sourceURL, String fileName) {
+def deobf(String sourceURL, String rawFileName) {
+    String bon2Version = "2.5.1"
+    String fileName = URLDecoder.decode(rawFileName, "UTF-8")
     String cacheDir = System.getProperty("user.home") + "/.gradle/caches/"
     String bon2Dir = cacheDir + "forge_gradle/deobf"
-    String bon2File  = bon2Dir + "/BON2-2.5.0.jar"
+    String bon2File  = bon2Dir + "/BON2-" + bon2Version + ".jar"
     String obfFile = cacheDir + "modules-2/files-2.1/" + fileName + ".jar"
     String deobfFile = cacheDir + "modules-2/files-2.1/" + fileName + "-deobf.jar"
 
     if(file(deobfFile).exists()) {
         return files(deobfFile)
     }
 
+    checkPropertyExists("channel")
+    checkPropertyExists("mappingsVersion")
+    checkPropertyExists("remoteMappings")
+    String mappingsVer
+    if(remoteMappings) {
+        String id = forgeVersion.split("\\.")[3] + '-' + minecraftVersion
+        String mappingsZIP = cacheDir + "/forge_gradle/maven_downloader/de/oceanlabs/mcp/mcp_snapshot_nodoc/" + id + "/mcp_snapshot_nodoc-" + id + ".zip"
+
+        zipMappings(mappingsZIP, remoteMappings, bon2Dir)
+        
+        mappingsVer = 'snapshot_' + id
+    } else {
+        mappingsVer = channel + '_' + mappingsVersion
+    }
+
     download.run {
-        src 'https://github.com/GTNewHorizons/BON2/releases/download/2.5.0/BON2-2.5.0.CUSTOM-all.jar'
+        src 'http://jenkins.usrv.eu:8081/nexus/content/repositories/releases/com/github/parker8283/BON2/' + bon2Version + '-CUSTOM/BON2-' + bon2Version + '-CUSTOM-all.jar'
```

Comment:
I *think* because this is on maven, you can use gradle's regular dependency handling to fetch the jar, and make a special task type to execute it referring to that dependency instead of implementing it by hand, see examples at <https://discuss.gradle.org/t/running-an-executable-jar-from-a-remote-maven-repo/15444>